selendroid-standalone/src/main/java/io/selendroid/standalone/builder/SelendroidServerBuilder.java

```diff
@@ -364,22 +360,77 @@ public static String getJarVersionNumber() {
     Class clazz = SelendroidStandaloneDriver.class;
     String className = clazz.getSimpleName() + ".class";
     String classPath = clazz.getResource(className).toString();
+    String version = "";
+
     if (!classPath.startsWith("jar")) {
       // Class not from JAR
-      return "dev";
-    }
-    String manifestPath =
-        classPath.substring(0, classPath.lastIndexOf("!") + 1) + "/META-INF/MANIFEST.MF";
-    Manifest manifest;
-    try {
-      manifest = new Manifest(new URL(manifestPath).openStream());
-    } catch (Exception e) {
-      return "";
+      String dev_path = "";
+      if(classPath.startsWith("file") && classPath.contains("target"))
+        dev_path = classPath.substring(5, classPath.lastIndexOf("target"));
+      try {
+        version = getVersionFromPom(dev_path);
+      } catch (Exception e) {
+        e.printStackTrace();
+        return "pom file path may not exist";
+      }
+    } else {
+        try {
+          version = getVersionFromManifest(classPath);
+        } catch (IOException e){
+          return "";
+        }
     }
+    return version;
+  }
+
+  private static String getVersionFromManifest(String path) throws IOException {
+    String manifestPath = path.substring(0, path.lastIndexOf("!") + 1) + "/META-INF/MANIFEST.MF";
+    Manifest manifest = new Manifest(new URL(manifestPath).openStream());
     Attributes attr = manifest.getMainAttributes();
     return attr.getValue("version");
   }
 
+  private static String getVersionFromPom(String path) throws Exception {
+    String version = "";
+    path += "pom.xml";
+    String fileContents = getFileContents(path);
+
+    Pattern startPattern = Pattern.compile("<version>");
+    Matcher startMatch = startPattern.matcher(fileContents);
+
+    while(startMatch.find()) {
```

Comment:
i think using the regex pattern as shown in the link and getting the group (the value inbetween the tags, because it's surrounded by parenthesis) is a cleaner approach.

So, something like:
Pattern regex = Pattern.compile("<version>(.*?)</version>", Pattern.DOTALL);
Matcher matcher = regex.matcher(FileUtils.readFileToString(new File(path)));
if (matcher.find()) {
    return matcher.group(1);
}